packages/clerk-js/src/core/clerk.ts

```diff
@@ -523,16 +532,26 @@ export default class Clerk implements ClerkInterface {
   };
 
   public authenticateWithMetamask = async ({ redirectUrl }: AuthenticateWithMetamaskParams = {}): Promise<void> => {
-    if (!this.client) {
+    if (!this.client || !this.#environment) {
       return;
     }
 
+    const { signUpUrl } = this.#environment.displayConfig;
+
     let signInOrSignUp: SignInResource | SignUpResource;
     try {
       signInOrSignUp = await this.client.signIn.authenticateWithMetamask();
     } catch (err) {
       if (isError(err, ERROR_CODES.FORM_IDENTIFIER_NOT_FOUND)) {
         signInOrSignUp = await this.client.signUp.authenticateWithMetamask();
+
+        if (
+          signInOrSignUp.verifications.web3Wallet.status === 'verified' &&
+          signInOrSignUp.status === 'missing_requirements'
+        ) {
+          // @ts-ignore
+          return this.navigate(signUpUrl + '/continue');
```

Comment:
🔧 Let's use `navigateToContinueSignUp`

Reply:
I think this has the same hash routing issue Colin mentioned in a previous comment.

Reply:
❓ Why do we need ts-ignore here?

Reply:
@nikosdouvlis 
The return type of navigate is not `void` . We can go with:
`void this.navigate(signUpUrl + '/continue');` or just not return